README.md

```diff
@@ -1,16 +1,120 @@
+------------------------------------------------------------------------
 
 ## Description
 
+This package implements a numerical solver for the **2D thermal
+diffusion equation** on a square domain.\
+Initially, the plate is set to a uniform cold temperature, except for a
+hot circular region at the center.\
+The temperature then evolves according to the diffusion equation:
+
+$$
+\frac{\partial T}{\partial t}
+= D \left(
+\frac{\partial^2 T}{\partial x^2} +
+\frac{\partial^2 T}{\partial y^2}
+\right)
+$$
+
+### Main Features
+
+-   Explicit finite-difference scheme\
+-   Square domain with configurable grid spacing (`dx`, `dy`)\
+-   User-adjustable thermal diffusivity (`D`)\
+-   Visualizes the diffusion over time by generating **four heatmaps**
+    at selected time steps\
+-   Cleanly separated solver (`solve`) and plotting utilities
+    (`output.py`)\
+-   Fully packaged and deployable on TestPyPI
+
+### Reference Source
+
+The implementation is based on the example from:
+
+> *Learning Scientific Programming with Python*, Chapter 7\
+> https://scipython.com/book/chapter-7-matplotlib/examples/the-two-dimensional-diffusion-equation/
+
+------------------------------------------------------------------------
+
 ## Installing the package
 
+After uploading the distribution to **TestPyPI**, the package can be
+installed using:
+
+``` bash
+pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ rongchzg_diffusion2d
+```
+
+Explanation: - `test.pypi.org` contains your package\
+- `pypi.org` provides dependencies such as NumPy and Matplotlib
+
+------------------------------------------------------------------------
+
 ## Running this package
 
+### Option 1 --- Use the installed package
+
+``` python
+from rongchzg_diffusion2d import solve
+
+# Run with default parameters
+solve()
+
+# Run with custom parameters
+solve(dx=0.05, dy=0.05, D=2.0)
+```
+
+### Option 2 --- Run module directly (when cloned locally)
+
+``` bash
+python -m rongchzg_diffusion2d.diffusion2d
+```
+
+### What the solver does
+
+-   Initializes the domain with a hot circular region at the center\
+-   Computes time evolution using explicit finite differences\
+-   Produces **four subplots** at time steps `[0, 10, 50, 100]`\
+-   Shows the diffusion process visually through heatmaps
+
+------------------------------------------------------------------------
+
 ## Citing
 
+If this package is used in coursework, exercises, or academic work,
+please cite it as:
+
+    Zhang, Rongchuan. (2025). rongchzg_diffusion2d:
+    A 2D diffusion equation solver. TestPyPI Package.
+
+Or using BibTeX:
+
+``` bibtex
+@misc{rongchzg_diffusion2d_2025,
+  author = {Zhang, Rongchuan},
+  title  = {rongchzg\_diffusion2d: A 2D diffusion equation solver},
+  year   = {2025},
+  howpublished = {\url{https://test.pypi.org/project/rongchzg_diffusion2d/}}
+}
+```
```

Comment:
It was expected that the book chapter from which the original code is taken would be put here as a citation entity.